src/routes/discussions/index.js

```diff
@@ -86,30 +86,42 @@ const deleteById = async (req, res, next) => {
 };
 
 const getCommentsById = async (req, res) => {
-  const { id } = req.params;
   try {
+    const { discussionSlug } = req.params;
+    const discussion = await DiscussionService.getDiscussion(discussionSlug);
+
     const parents = await models.comments.findAll({
       where: {
-        discussion_id: id,
+        discussion_id: discussion.id,
       },
       include: {
         model: models.users,
         as: 'user',
       },
     });
 
-    let isLastStep = false; let
-      childs = await models.comments.findAll({
-        where: {
-          parent_comment_id: {
-            [Op.or]: parents.map((c) => c.id),
+    if (parents.length === 0) {
+      return res.send({
+        errors: [
```

Comment:
Burada kullanıcıya hata mesajı dönmek yerine boş dizi dönmek frontend için daha uygun bir senaryo olabilir.